tools/protoprint/protoprint.py

```diff
@@ -726,20 +747,22 @@ def visit_file(self, file_proto, type_context, services, msgs, enums):
         return clang_format(header + formatted_services + formatted_enums + formatted_msgs)
 
 
-if __name__ == '__main__':
-    proto_desc_path = sys.argv[1]
+class ProtoprintTraverser:
+
+    def traverse_file(self, file_proto, visitor):
+        # This breaks the Visitor contract, ie class props should not be set on individual visitation.
```

Comment:
this reflects what was being done already in the `main` function and is i think the reason why this was not implemented as a protoc plugin originally - either way this is necessary without making other changes to traverser/visitor

Reply:
Yeah, this is kind of evil. Can you add some explanation of "why" for future readers?

Reply:
todo added